src/Twig/Extension/PaginationRuntime.php

```diff
@@ -109,20 +109,22 @@ public function filter(
     }
 
     /**
-     * @param array<string, mixed> $query
-     * @param int $page
+     * @param array<string, mixed>      $query
+     * @param int                       $page
+     * @param array<string, mixed>|null $paginatorOptions
      * @return array<string, mixed>
      */
-    public function getQueryParams(array $query, int $page): array
+    public function getQueryParams(array $query, int $page, ?array $paginatorOptions = []): array
     {
+        $pageName = $paginatorOptions['pageParameterName'] ?? $this->pageName;
         if ($page === 1 && $this->skipFirstPageLink) {
-            if (isset($query[$this->pageName])) {
-                unset($query[$this->pageName]);
+            if (isset($query[$pageName])) {
+                unset($query[$pageName]);
             }
 
             return $query;
         }
 
-        return array_merge($query, [$this->pageName => $page]);
+        return array_merge($query, [$pageName => $page]);
```

Comment:
this phpstan failure needs to be addressed

Reply:
I don't understand how the change which has been made can be the cause of this error? `$pageName` will be a string, otherwise nothing about the array_merge has changed?

Reply:
The variable now can come from the options array, so it can be different from before

Reply:
I'm not familiar with PHPStan, however I presume that now that the `$options` parameter for the method is no-longer nullable, and that it is defined as `@param array<string, mixed>` the return of `@return array<string, mixed>` will be accepted by PHPStan? 

If not, then what is your preferred method of handling this? Should I change the return to be `@return array<int|string, mixed>` or explicitly case the `$pageName` variable to a string? Or?

Reply:
The problem is that we declare the options as `mixed`, so there's no guarantee that the value of `$options['pageParameterName']` is a string.

This should work:
```
$pageName = $this->pageName;
if (isset($options['pageParameterName']) && is_string($options['pageParameterName'])) {
    $pageName = $options['pageParameterName'];   
}
```

Reply:
Thanks  I've implemented as suggested, let us hope that it is happy now.